pkg/operator/rbaccontroller/rbaccontroller.go

```diff
@@ -0,0 +1,102 @@
+package rbaccontroller
+
+import (
+	"context"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	listerrbacv1 "k8s.io/client-go/listers/rbac/v1"
+	"k8s.io/klog"
+)
+
+const (
+	// On an upgrade from 4.1, some discovery cluster role bindings would
```

Comment:
this description sounds very specific to the issue we had. That's good as background. But we need a crisp definition of what it does in general.